### PR TITLE
Provide local CSS when external CDN fails

### LINKED ISSUE #352: Add fallback code for libraries loaded via CDN
In the case where a library cannot be loaded from a CDN, the code should be loaded locally. Something like the code included [here](http://www.hanselman.com/blog/CDNsFailButYourScriptsDontHaveToFallbackFromCDNToLocalJQuery.aspx) would probably work.

### PR DESCRIPTION
**Pull request by ryanjohara**
Fixes #352

Signed-off-by: Ryan O'Hara oharar@email.chop.edu

### CODE REVIEW

**Comment by naegelyd** [comment]
Please rebase this on `master`. I had to muck with the revision history because I apparently can't increment versions today....

File: varify/templates/base.html [hunk 1]
```diff
@@ -16,6 +16,35 @@
...
+                        cssFound = true;
+                    }
+                }
+                return cssFound;
+            }
+
+            var cssStyles = ['bootstrap.css','bootstrap-responsive.css','font-awesome.css'];
+
+            for (var cs in cssStyles) {
+
```
Comment (naegelyd) [inline varify/templates/base.html:37]: Remove blank line

File: varify/templates/base.html [hunk 2]
```diff
@@ -16,6 +16,35 @@
...
+                    var sheet = document.styleSheets[i];
+
+                    if (sheet['href'].indexOf(href) >= 0) {
+                        cssFound = true;
+                    }
+                }
+                return cssFound;
+            }
+
+            var cssStyles = ['bootstrap.css','bootstrap-responsive.css','font-awesome.css'];
```
Comment (naegelyd) [inline varify/templates/base.html:34]: Spaces after commas

File: varify/templates/base.html [hunk 3]
```diff
@@ -16,6 +16,35 @@
...
+            function cssLoaded(href) {
+                var cssFound = false;
+
+                for (var i = 0; i < document.styleSheets.length; i++) {
+                    var sheet = document.styleSheets[i];
+
+                    if (sheet['href'].indexOf(href) >= 0) {
+                        cssFound = true;
+                    }
+                }
```
Comment (naegelyd) [inline varify/templates/base.html:30]: Add blank line below

File: varify/templates/base.html [hunk 4]
```diff
@@ -16,6 +16,35 @@
...
+                return cssFound;
+            }
+
+            var cssStyles = ['bootstrap.css','bootstrap-responsive.css','font-awesome.css'];
+
+            for (var cs in cssStyles) {
+
+                if (!cssLoaded(cssStyles[cs])) {
+                    var local_bootstrap = document.createElement('link');
+                    local_bootstrap.setAttribute("rel", "stylesheet");
```
Comment (naegelyd) [inline varify/templates/base.html]: Use single quotes here and everywhere else below

**Comment by naegelyd** [comment]
@ryanjohara Have you tested if the fallbacks work by giving a bad path for bootstrap or something to make sure that the local copy gets used?

**Comment by ryanjohara** [comment]
Yes, I tested with an invalid CDN URL before I committed.

**Comment by naegelyd** [comment]
Great, just wanted to be sure

**Pull request closed without merging**